semconv-incubating/src/main/java/io/opentelemetry/semconv/incubating/HttpIncubatingAttributes.java

```diff
@@ -121,10 +121,10 @@ public final class HttpIncubatingAttributes {
    * </ul>
    *
    * @deprecated deprecated in favor of stable {@link
-   *     io.opentelemetry.semconv.HttpAttributes#HTTP_REQUEST_RESEND_COUNT} attribute.
+   *     io.opentelemetry.semconv.HttpAttributes#HTTP_REQUEST_RESENDCOUNT} attribute.
    */
   @Deprecated
-  public static final AttributeKey<Long> HTTP_REQUEST_RESEND_COUNT =
+  public static final AttributeKey<Long> HTTP_REQUEST_RESENDCOUNT =
       longKey("http.request.resend_count");
```

Comment:
what do you think about alternative proposal:
* first one gets the good constant name: `MESSAGING_CLIENT_ID = "messaging.client_id"`
* next one (after rename) gets a "bad" constant name: `MESSAGING_CLIENT_ID2 = "messaging.client.id"`
* when stabilizing, we are moving from `MessagingIncubatingAttributes` to `MessagingAttributes` and so the stable one can get the good constant name at that time: `MESSAGING_CLIENT_ID = "messaging.client.id"`
* if anything is renamed after stability, we'd need to do a major version bump anyways (although users of the previously stable constant name would silently get the new constant value without any compilation warnings, which is not great, but really hoping this kind of rename after marking something stable is a rarity 😬)

Reply:
I prefer this plan.

Reply:
we could also just tolerate it:
- generate both as MESSAGING_CLIENT_ID, override the value (non-deprecated wins) - it's tolerable for experimental attributes
- as you mentioned, we'd probably never do a rename like this for a stable attribute - there value/cost would be too low.

Reply:
I'm not excited about `MESSAGING_CLIENT_ID2` since it would either need to be stabilized with this const name or users would need to update `IncubatingAttributes.MESSAGING_CLIENT_ID2`  -> `Attributes.MESSAGING_CLIENT_ID`

Reply:
> or users would need to update `IncubatingAttributes.MESSAGING_CLIENT_ID2` -> `Attributes.MESSAGING_CLIENT_ID`

this doesn't seem too bad to me

Reply:
So 
v1.25.0: 
```java
MESSAGING_CLIENT_ID = "messaging.client_id";
```

v1.26.0: 
```java
MESSAGING_CLIENT_ID = "messaging.client.id";
MESSAGING_CLIENT_ID_DEPRECATED = "messaging.client_id";
```

?

I have no objections, it just feels unnecessary maybe.
I.e. as a user, I update to the new version of semconv
- I need to discover what's changed
- Change `MESSAGING_CLIENT_ID` to `MESSAGING_CLIENT_ID_DEPRECATED`
- If I don't do it, I still get "messaging.client.id" 

So it allows user to keep using old attribute but only if they do some extra work. Which a few users will and we probably want them to update the semconv version in schema url instead.

Reply:
ah, sorry, this is what I was thinking:

v1.25.0: 
```java
MESSAGING_CLIENT_ID = "messaging.client_id";
```

v1.26.0: 
```java
MESSAGING_CLIENT_ID = "messaging.client_id";
MESSAGING_CLIENT_ID2 = "messaging.client.id";
```

Reply:
What  would we put into stable artifact when it's stabilized? `MESSAGING_CLIENT_ID`?

So when I'm updating my code to use stable attribute names, I'd change `MessagingIncubatingAttributes.MESSAGING_CLIENT_ID2` -> `MessagingAttributes.MESSAGING_CLIENT_ID`.

`MessagingIncubatingAttributes.MESSAGING_CLIENT_ID2` should be annotated with `@deprecated("Use {@link saemconv.MessagingAttributes.MESSAGING_CLIENT_ID")`

This seems to be ok - I need to do code changes anyway.

Reply:
The problem still remains - we don't really know the order in which attributes were added and which one should become `MESSAGING_CLIENT_ID2`. We can assume that deprecated should get `MESSAGING_CLIENT_ID`, but what if we deprecated both?

Reply:
> What would we put into stable artifact when it's stabilized? `MESSAGING_CLIENT_ID`?

👍

Reply:
we can do it with some manual interventions. 
Given  that collisions are rare, it should be ok. 

Trying it out here https://github.com/open-telemetry/semantic-conventions-java/pull/76/commits/63908477f1cecff11a36381a34041a4199c1cbee

Reply:
I don't mind the approach 3.5 where we accept breaking changes in the incubating artifact, but wondering if this solution could work better for folks who are more apprehensive about the breaking changes (and the strategy could also be used to mitigate the even rarer attribute type changes on experimental attributes)

Reply:
https://github.com/open-telemetry/semantic-conventions-java/pull/76 shows manual conflict resolution:

- it allows to change constant names for experimental attributes. E.g. say that `messaging.client.id` constant names should be generated as if it was `messaging.client.id_new`
- or drop an attribute based on the name